pkg/jsonAPI/jsonapi.go

```diff
@@ -0,0 +1,514 @@
+		},
+	}
+
+	return &summary
+}
+
+// stats on failure groups
+func jsonFailureGroups(failureGroups, failureGroupsPrev []util.JobRunResult, endDay int) *FailureGroups {
+	count, countPrev, median, medianPrev, avg, avgPrev := 0, 0, 0, 0, 0, 0
+	for _, group := range failureGroups {
+		count += group.TestFailures
+	}
+	for _, group := range failureGroupsPrev {
+		countPrev += group.TestFailures
+	}
+	if len(failureGroups) != 0 {
+		median = failureGroups[len(failureGroups)/2].TestFailures
+		avg = count / len(failureGroups)
+	}
+	if len(failureGroupsPrev) != 0 {
+		medianPrev = failureGroupsPrev[len(failureGroupsPrev)/2].TestFailures
+		avgPrev = count / len(failureGroupsPrev)
+	}
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+
+	failureGroupStruct := FailureGroups{
+		JobRunsWithFailureGroup: map[string]int{
+			latestDays: len(failureGroups),
+			prevDays:   len(failureGroupsPrev),
+		},
+		AvgFailureGroupSize: map[string]int{
+			latestDays: avg,
+			prevDays:   avgPrev,
+		},
+		MedianFailureGroupSize: map[string]int{
+			latestDays: median,
+			prevDays:   medianPrev,
+		},
+	}
+	return &failureGroupStruct
+}
+
+func jsonSummaryJobsByPlatform(report, reportPrev util.TestReport, endDay, jobTestCount int) []JobSummaryPlatform {
+	jobsByPlatform := util.SummarizeJobsByPlatform(report)
+	jobsByPlatformPrev := util.SummarizeJobsByPlatform(reportPrev)
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+
+	var jobSummariesByPlatform []JobSummaryPlatform
+
+	for _, v := range jobsByPlatform {
+		prev := getPrevPlatform(v.Platform, jobsByPlatformPrev)
+
+		var jobSummaryPlatform JobSummaryPlatform
+
+		if prev != nil {
+			jobSummaryPlatform = JobSummaryPlatform{
+				Platform: v.Platform,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage:          v.PassPercentage,
+						ProjectedPercentage: v.PassPercentageWithKnownFailures,
+						Runs:                v.Successes + v.Failures,
+					},
+					prevDays: PassRate{
+						Percentage:          prev.PassPercentage,
+						ProjectedPercentage: prev.PassPercentageWithKnownFailures,
+						Runs:                prev.Successes + prev.Failures,
+					},
+				},
+			}
+		} else {
+			jobSummaryPlatform = JobSummaryPlatform{
+				Platform: v.Platform,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage:          v.PassPercentage,
+						ProjectedPercentage: v.PassPercentageWithKnownFailures,
+						Runs:                v.Successes + v.Failures,
+					},
+				},
+			}
+		}
+
+		platformTests := report.ByPlatform[v.Platform]
+		for _, test := range platformTests.TestResults {
+			if util.IgnoreTestRegex.MatchString(test.Name) {
+				continue
+			}
+
+			encodedTestName := url.QueryEscape(regexp.QuoteMeta(test.Name))
+			// NOTE: not really sure what this represents
+			// jobQuery := fmt.Sprintf("%s.*%s|%s.*%s", report.Release, v.Platform, v.Platform, report.Release)
```

Comment:
it's part of the ci-search query that filters the query down to just job names that match the job name we're listing test failures for.  

your `testLink` below should include this value as the `name` query parameter.  see:
https://github.com/openshift/sippy/blob/master/pkg/html/html.go#L310
https://github.com/openshift/sippy/blob/master/pkg/html/html.go#L322-L329

which is populating:
https://github.com/openshift/sippy/blob/master/pkg/html/html.go#L128

It allows someone to see a search result that only shows "failures of this specific test, when run in jobs that match this specific platform"

i think your "testLink" url should include it in the same way since this is a "by platform" set of results you're building.